File: java/01-commands/doc/sourcecode/src/test/java/com/training/app/AccountStatementPrintTest.java
package com.training.app;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class AccountStatementPrintTest {

    @Test
    public void whenAPrintStatementIsAskedAfterTheFirstDeposit_DateAmountAndBalanceArePrinted() {
        // Arrange 
        List<BankingTransactionDTO> statement = new ArrayList<>();
        String expectedOutput = "Date Amount Balance\n" +
        LocalDate.now().toString() + " +500 500\n";
        
        // Act
        statement.add(new BankingTransactionDTO(LocalDate.now(), 500, 500));

        // Assert
        assertEquals(expectedOutput, AccountStatementPrint.printStatement(statement));
    }

    @Test
    public void whenAPrintStatementIsAskedAfterADepositAndAWithdraw_DateAmoutAndBalanceArePrinted() {
        // Arrange 
        List<BankingTransactionDTO> statement = new ArrayList<>();
        String expectedOutput = "Date Amount Balance\n" +
        LocalDate.now().toString() + " +500 500\n" +
        LocalDate.now().toString() + " -100 400\n";

        // Act
        statement.add(new BankingTransactionDTO(LocalDate.now(), 500, 500));
        statement.add(new BankingTransactionDTO(LocalDate.now(), -100, 400));

        // Assert
        assertEquals(expectedOutput, AccountStatementPrint.printStatement(statement));
    }
}

File: python/01-commands/doc/Banking_Kara/account_statement_print_test.py
import unittest
from datetime import datetime
from banking_transaction_dto import BankingTransactionDTO
from account_statement_print import print_statement

class AccountStatementPrintTest(unittest.TestCase):
    def test_print_statement_after_first_deposit(self):
        # Arrange
        statement = []
        expected_output = "Date Amount Balance\n" + \
                          f"{datetime.today().strftime('%m/%d/%Y')} +500 500\n"

        # Act
        statement.append(BankingTransactionDTO(datetime.today(), 500, 500))

        # Assert
        self.assertEqual(expected_output, print_statement(statement))

    def test_print_statement_after_deposit_and_withdraw(self):
        # Arrange
        statement = []
        expected_output = "Date Amount Balance\n" + \
                          f"{datetime.today().strftime('%m/%d/%Y')} +500 500\n" + \
                          f"{datetime.today().strftime('%m/%d/%Y')} -100 400\n"

        # Act
        statement.append(BankingTransactionDTO(datetime.today(), 500, 500))
        statement.append(BankingTransactionDTO(datetime.today(), -100, 400))

        # Assert
        self.assertEqual(expected_output, print_statement(statement))

if __name__ == '__main__':
    unittest.main()